library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+// Create a new Envoy instance. The Envoy runner NSThread is started as part of instance
+// initialization with the configuration provided.
+- (id)initWithConfig:(NSString *)config {
+  self.runner = [[NSThread alloc] initWithTarget:self selector:@selector(run:) object:config];
```

Comment:
Would it make sense to move these to the common `init` from `NSObject` so we can avoid the null checks below?

Reply:
Hm. I guess maybe if we had a default config we could pass through?

Reply:
At this point, not sure what would make sense there though, tbh.

Reply:
Good here? Since obj-c magics away the nil checks for us anyways.

Reply:
Yea that's fine